data/users.js

```diff
@@ -7,470 +7,519 @@ const bcryptjs = require("bcryptjs");
-            throwError(ErrorCode.NOT_FOUND, "Error: User not found.");
-        }
-
-        return user;
-    } catch (error) {
-        throwCatchError(error);
+  try {
+    validateGetTotalArguments(arguments.length);
+
+    const userId = validateUserId(xss(_userId));
+
+    const usersCollection = await users();
+
+    const user = await usersCollection.findOne(
+      { _id: userId },
+      {
+        projection: {
+          _id: 1,
+          firstName: 1,
+          lastName: 1,
+          email: 1,
+          dateOfBirth: 1,
+        },
+      }
+    );
+
+    if (!user) {
+      throwError(ErrorCode.NOT_FOUND, "Error: User not found.");
     }
+
+    return user;
+  } catch (error) {
+    throwCatchError(error);
+  }
 }
 
 async function getAll() {
-    try {
-        const usersCollection = await users();
+  try {
+    const usersCollection = await users();
+
+    const usersList = await usersCollection.find({}).toArray();
 
-        const usersList = await usersCollection.find({}).toArray();
+    return usersList;
+  } catch (error) {
+    throwCatchError(error);
+  }
+}
 
-        return usersList;
-    } catch (error) {
-        throwCatchError(error);
+async function getUserStatistics(_userId) {
+  /*
+    - Total number of tasklists
+    - Total number of tasks across all tasklists
+    - Number of tasks completed on time
+    - Number of tasks not completed on time
+    - Number of tasks completed overall
+    - Number of tasks not completed overall
+  */
+  validateGetTotalArguments(arguments.length);
+
+  const userId = validateUserId(_userId);
+
+  const tasklistsCollection = await tasklists();
+
+  const tasklists = tasklistsCollection.find({ userId: userId }).toArray();
+  const numTasklists = tasklists.length;
+
+  const tasksCollection = await tasks();
+  const tasks = tasksCollection.find({ userId: userId }).toArray();
+  const numTasks = tasks.length;
+
+  let completedOnTime;
+  let notCompletedOnTime;
+  let completed;
+  let notCompleted;
+
+  for (let task of tasks) {
+    let deadlineDate = task.deadlineDate;
+    let completionDate = task.completionDate;
+    // Converting both to MM/DD/YYYY format just in case
+    let deadline = moment(Date.parse(deadlineDate)).format("MM/DD/YYYY");
+    let completion = moment(Date.parse(completionDate)).format("MM/DD/YYYY");
+    let isCompleted = task.isCompleted;
+
+    if (isCompleted) {
+      completed += 1;
+
+      let doneOnTime = moment(Date.parse(completion)).isBefore(
+        Date.parse(deadline)
+      );
+      if (doneOnTime) {
+        completedOnTime += 1;
+      } else {
+        notCompletedOnTime += 1;
+      }
+    } else {
+      notCompleted += 1;
     }
+
+    let userStatistics = {
+      numTasklists: numTasklists,
+      numTasks: numTasks,
+      completedOnTime: completedOnTime,
+      notCompletedOnTime: notCompletedOnTime,
+      completed: completed,
+      notCompleted: notCompleted,
+    };
+
+    return userStatistics;
+  }
+  try {
```

Comment:
`try{}` block is empty. Please put entire code inside `try{}`

Reply:
whoops, code moved around when i pulled main branch into dev branch, thanks for catching that